### ISSUE TITLE
Unclear install instructions for cider (instead of nrepl)

### ISSUE THREAD

**Issue by clojj** [coventry/troncle#3]
Couldn't make it work with cider.

loading errors in emacs when starting.

Maybe, if it is already working with newest cider-version, you can document the installation procedure with cider ?

**Comment by coventry** [comment]
Thanks for reporting this.  If you're getting errors on starting, can you please send me the result of "emacs --debug-init"?

**Comment by bbatsov** [comment]
The problem is that the package depends on nrepl, but nrepl and cider cannot co-exist, since some functions and variables are present in both packages. I guess one workaround would be removing the nrepl dependency as well, so people can simply use whatever package they have installed. I'd recommend dropping support for the legacy nrepl package once cider 0.4 is out.

**Comment by coventry** [comment]
@joergwin thanks again.  I was  able to reproduce the problem.  Could you please try putting the version at https://github.com/coventry/troncle/blob/master/troncle.el on your `load-path` and let me know the results?

If you installed from marmalade, please uninstall it, and use this version directly.  Marmalade is down, so I can't fix it there right now.

**Comment by coventry** [comment]
@bbatsov Could you please give me some more detail on the problem the nrepl dependency is causing?  It's not causing a problem with cider, in my hands.

**Comment by bbatsov** [comment]
@coventry The problem is that there are some functions with the same names (in cider they are in `nrepl-client.el`), but different code in the nrepl 0.2 package and the current cider package. If you have nrepl and cider installed the one that's loaded second would overwrite those function definitions from the package that was loaded first and cause problems for packages relying on the first. That's why I'm urging everyone to simply drop support for the old package and move on - life will be easier for both the developers and the users that way. Another problem with the current situation is that troncle cannot be added to MELPA while it depends on nrepl.el and I can assure you that more people are using MELPA than Marlamade these days.

On a related note - you might consider moving your project over to `clojure-emacs` to increase its visibility in the Clojure/Emacs hacker community.

**Comment by clojj** [comment]
@coventry thanks, will try it asap

**Comment by clojj** [comment]
sorry, where is load-path ?
I tried copying that troncle.el into /usr/local/share/emacs/24.3/site-lisp/

and (load-library "troncle") into the file ~/.emacs   (?)

won't work

'M-x troncle-set-exec-var' is not available.

BTW am usng latest cider version from MELPA, no nrepl installed !

here is output from emcs --debug-init

Debugger entered--Lisp error: (file-error "Cannot open load file" "nrepl")
  require(nrepl)
  (if (require (quote cider) nil t) nil (require (quote nrepl)))
  eval-buffer(#<buffer  *load*-791779> nil "/usr/local/share/emacs/24.3/site-lisp/troncle.el" nil t)  ; Reading at buffer position 665
  load-with-code-conversion("/usr/local/share/emacs/24.3/site-lisp/troncle.el" "/usr/local/share/emacs/24.3/site-lisp/troncle.el" nil nil)
  load("troncle")
  load-library("troncle")
  eval-buffer(#<buffer  *load*> nil "/home/jwin/.emacs" nil t)  ; Reading at buffer position 25
  load-with-code-conversion("/home/jwin/.emacs" "/home/jwin/.emacs" t t)
  load("~/.emacs" t t)
  #[0 "\205\262

**Comment by coventry** [comment]
Hi, joergwin.  Thanks for the traceback.  Could you please tell me the output of `(require (quote cider) nil t)`?  Go to the emacs `*scratch*` buffer, paste that form in, put the cursor inside it and type `C-M-x`.  The result should appear in the minibuffer for a moment.  If you miss it, it'll be in the `*Messages*` buffer.  For me in my cider testbed the result is `cider`, but it looks as though in that traceback the result was `nil`.

**Comment by clojj** [comment]
Hi, I just pasted all of the troncle.el to the end of my vanilla init.el
that way, it works

(before, I installed the current cider in a fresh /.emacs.d)

I'm an emacs newbie, so it's probably also possible in another way.

**Comment by coventry** [comment]
Hi, joergwin.  I'm glad you got it working, but I would like to make this process friendlier for cider users.  It would be great to reproduce the failure you described above.  Would you mind sending me your emacs configuration files?  Something like 

``` bash
cd && tar zcf emacs.tgz .emacs*
```

should produce an `emacs.tgz` file which you could send me as an attachment to my github username at gmail.com.

**Comment by ghost** [comment]
@coventry, I hit the same problem. I've got cider installed (20131221.1231) and 
`package-install troncle` failed for me. It fetched nrepl-0.2.0 and troncle-0.12. into `elpa/`but bugged out
with an error and the installation didn't complete (an error about invalid parens , not sure where from).
Since I already have cider setup, forcing an nrepl potentially borks my system.

I ended up putting troncle in the load-path, requiring it and things worked then. I'll open a new issue
with some suggestions.
I really wish the transition period for cider was over, half the emacs packages I've sampled
just don't work properly.

**Comment by bbatsov** [comment]
@y-p As I said earlier - the `nrepl` dep needs to be dropped and replaced with a `cider` dep.

**Comment by ghost** [comment]
I was hoping for just that.

**Comment by coventry** [comment]
Thanks for reporting this, @y-p.  I have changed the dependency as suggested, and uploaded a new troncle 0.1.2.  Please delete the old package and reinstall, and let me know if you run into any difficulties.

**Comment by ghost** [comment]
Yes, it works fine now. -pkg.el lists cider 0.5.0 as min version, 0.4.0 might be a better choice.
Also, no warning to the user if the middleware isn't installed (I changes profiles around and accidentaly
removed it), would be nice to get a warning if it's doable.

Thanks!
